### PR TITLE
Update templates to Roslyn 3

### PR DESCRIPTION
**Pull request by sharwell**
Template updates:

* Use `<SuppressDependenciesWhenPacking>` to simplify package references
* Convert VSIX deployment projects to CPS
* Update templates to Roslyn 3 references

Infrastructure updates:

* Show all template files in Solution Explorer

### CODE REVIEW

File: src/VisualStudio.Roslyn.SDK/Roslyn.SDK/ProjectTemplates/CSharp/CodeRefactoring/Vsix/Deployment.csproj [hunk 1]
```diff
@@ -1,23 +1,14 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project>
+  <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.props" />
```
Comment (jmarolf) [inline src/VisualStudio.Roslyn.SDK/Roslyn.SDK/ProjectTemplates/CSharp/CodeRefactoring/Vsix/Deployment.csproj]: Generally a fan of this. Are there any known issues with VSIX projects in sdk-style projects? in the past there were bugs with the VSSDK around doing this.

Comment (sharwell) [reply src/VisualStudio.Roslyn.SDK/Roslyn.SDK/ProjectTemplates/CSharp/CodeRefactoring/Vsix/Deployment.csproj]: There are known issues, but they are addressed by the form of this project.

1. There are two explicit (linked) workarounds in the project file
2. The explicit `<Import>` elements are required to have everything defined in the correct order (i.e. the simplified `<Project Sdk="...">` form will not work for VSIX projects)

File: src/VisualStudio.Roslyn.SDK/Roslyn.SDK/ProjectTemplates/CSharp/CodeRefactoring/Ref/CodeRefactoring.csproj [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
   <PropertyGroup>
-    <TargetFramework>netstandard1.3</TargetFramework>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <SuppressDependenciesWhenPacking>true</SuppressDependenciesWhenPacking>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="2.6.2" PrivateAssets="all" />
-    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="2.10" PrivateAssets="all" />
-    <PackageReference Update="NETStandard.Library" PrivateAssets="all" />
```
Comment (jmarolf) [inline src/VisualStudio.Roslyn.SDK/Roslyn.SDK/ProjectTemplates/CSharp/CodeRefactoring/Ref/CodeRefactoring.csproj]: What dependecies does the resulting nupkg have with this change?

Comment (sharwell) [reply src/VisualStudio.Roslyn.SDK/Roslyn.SDK/ProjectTemplates/CSharp/CodeRefactoring/Ref/CodeRefactoring.csproj]: No dependencies at all. `SuppressDependenciesWhenPacking` even removes the list of declared target frameworks supported by the package.

**Pull request merged**